server/src/main/java/org/elasticsearch/health/node/UpdateHealthInfoCacheAction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health.node;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.health.node.action.TransportHealthNodeAction;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+
+public class UpdateHealthInfoCacheAction extends ActionType<AcknowledgedResponse> {
+
+    public static class Request extends ActionRequest {
+        private final String nodeId;
+        private final DiskHealthInfo diskHealthInfo;
+
+        public Request(String nodeId, DiskHealthInfo diskHealthInfo) {
+            this.nodeId = nodeId;
+            this.diskHealthInfo = diskHealthInfo;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.nodeId = in.readString();
+            this.diskHealthInfo = in.readOptionalWriteable(DiskHealthInfo::new);
+        }
+
+        public String getNodeId() {
+            return nodeId;
+        }
+
+        public DiskHealthInfo getDiskHealthInfo() {
+            return diskHealthInfo;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(nodeId);
+            out.writeOptionalWriteable(diskHealthInfo);
```

Comment:
Is this optional?

Reply:
Hm, no, on a second thought if it's missing we would use the unknown value never null.